test/src/kbasesearchengine/test/main/SearchMethodsTest.java

```diff
@@ -277,5 +285,136 @@ private void searchTypesCheckMatchFilter(
+        pp2.objectInfo = false;
+        pp2.objectDataIncludes=null;
+
+        final FoundHits fh1 = new FoundHits();
+        fh1.pagination = null;
+        fh1.sortingRules =  Arrays.asList(sr);
+        fh1.total = 1;
+        fh1.guids = set(new GUID("ws:1/2/3"));
+        fh1.objects = objs;
+
+        final FoundHits fh2 = new FoundHits();
+        fh2.pagination = null;
+        fh2.sortingRules =  Arrays.asList(sr);
+        fh2.total = 1;
+        fh2.guids = set(new GUID("ws:4/5/6"));
+        fh2.objects = objs2;
+
+        //result with highlight
+        when(idx.searchObjects(
+                new ArrayList<>(),
+                filter,
+                null, // sort
+                new kbasesearchengine.search.AccessFilter().withAccessGroups(set()),
+                null, // pagination
+                pp1))
+                .thenReturn(fh1);
+
+        //result with no highlight
+        when(idx.searchObjects(
+                new ArrayList<>(),
+                filter,
+                null, // sort
+                new kbasesearchengine.search.AccessFilter().withAccessGroups(set()),
+                null, // pagination
+                pp2))
+                .thenReturn(fh2);
+
+
+        //tests 
+        //with highlight on only
+        kbasesearchengine.PostProcessing option1 = new kbasesearchengine.PostProcessing()
+                .withIncludeHighlight(1L)
+                .withSkipKeys(1L)
+                .withSkipInfo(1L)
+                .withSkipData(1L);
+
+        final SearchObjectsOutput res1 = searchObjects(idx, option1);
+
+        final Map<String, List<String>> highlightRes = new HashMap<>();
+        highlightRes.put("field", highlight);
+        assertThat("did not find objects", res1.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res1.getObjects().get(0).getHighlight(), is(highlightRes));
+
+        //highlight on and ids on
+        kbasesearchengine.PostProcessing option2 = new kbasesearchengine.PostProcessing()
+                .withIncludeHighlight(1L)
+                .withIdsOnly(1L);
+
+        final SearchObjectsOutput res2 = searchObjects(idx, option2);
+
+        assertThat("did not find objects", res2.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res2.getObjects().get(0).getHighlight(), is(Collections.emptyMap()));
+
+
+        //highlight off and ids off
+        kbasesearchengine.PostProcessing option3 = new kbasesearchengine.PostProcessing()
+                .withIdsOnly(1L);
+
+        final SearchObjectsOutput res3 = searchObjects(idx, option3);
+
+        assertThat("did not find objects", res3.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res3.getObjects().get(0).getHighlight(), is(Collections.emptyMap()));
+
+    }
+
+    private SearchObjectsOutput searchObjects(final IndexingStorage idx, final kbasesearchengine.PostProcessing pp) throws Exception {
```

Comment:
\> 100 chars

Reply:
how do you eyeball this?!

Reply:
If it looks too long, I C&P into an editor and check